app/components/swipeable.js

```diff
@@ -72,17 +81,24 @@ export default class SwipeableComponent extends Component {
 
   doSwipe() {
     if (this.swiping) {
-      if (this.swipeDirection === 'horizontal') {
-        let direction = this.xEnd - this.xStart;
-        direction = direction > 0 ? 1 : -1;
-        this.args.onSwipeHorizontal(direction);
-      } else if (this.swipeDirection === 'vertical') {
-        let direction = this.yEnd - this.yStart;
-        direction = direction > 0 ? 1 : -1;
-        this.args.onSwipeVertical(direction);
+      let direction = 0;
+      if (this.horizontally) {
+        direction = this.xEnd - this.xStart;
+      } else if (this.vertically) {
+        direction = this.yEnd - this.yStart;
       } else {
-        throw new Error('No swipe direction defined');
+        this.throwNoDirection();
       }
+      direction = this.sign(direction);
+      this.onSwipe(direction);
     }
   }
+
+  sign(n) {
+    return n > 0 ? 1 : -1;
```

Comment:
Can't you use [`Math.sign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sign) here? This can return 0 though when you pass 0 into it but that's correct in case the touch ends at exactly the same location as it started I think

Reply:
good point!